src/components/Modals/SetAclPermsDocContainerModal.jsx

```diff
@@ -0,0 +1,172 @@
+import Dialog from '@mui/material/Dialog';
+import FormControl from '@mui/material/FormControl';
+import InputLabel from '@mui/material/InputLabel';
+import MenuItem from '@mui/material/MenuItem';
+import Select from '@mui/material/Select';
+import SettingsIcon from '@mui/icons-material/Settings';
+import TextField from '@mui/material/TextField';
+// Utility Imports
+import { runNotification, setDocContainerAclPermission } from '@utils';
+// Context Imports
+import { SignedInUserContext } from '@contexts';
+// Component Imports
+import FormSection from '../Form/FormSection';
+
+/**
+ * @typedef {import("../../typedefs.js").setAclPermsDocContainerModalProps} setAclPermsDocContainerModalProps
+ */
+
+/**
+ * SetAclPermsDocContainerModal Component - Modal component that generates the form for
+ * setting ACL permissions to another user's Documents container in their Solid
+ * Pod via Solid Session
+ *
+ * @memberof Forms
+ * @name SetAclPermsDocContainerModal
+ * @param {setAclPermsDocContainerModalProps} Props - Props for SetAclPermsDocContainerModal component
+ * @returns {React.JSX.Element} The SetAclPermsDocContainerModal Component
+ */
+const SetAclPermsDocContainerModal = ({ showModal, setShowModal }) => {
+  const { session } = useSession();
+  const { state, dispatch } = useStatusNotification();
+  const { podUrl } = useContext(SignedInUserContext);
+  const [permissionState, setPermissionState] = useState({
+    podUrlToSetPermissionsTo: '',
+    permissionType: ''
+  });
+
+  const clearInputFields = () => {
+    dispatch({ type: 'CLEAR_PROCESSING' });
+  };
+
+  const handleCloseModal = () => {
+    setPermissionState({
+      podUrlToSetPermissionsTo: '',
+      permissionType: ''
+    });
+    setShowModal(false);
+  };
+
+  // Event handler for setting ACL permissions to file container on Solid
+  const handleAclPermission = async (event) => {
+    event.preventDefault();
+    dispatch({ type: 'SET_PROCESSING' });
+    const permissions = event.target.setAclPerms.value
+      ? {
+          read: event.target.setAclPerms.value === 'Give',
+          append: event.target.setAclPerms.value === 'Give'
+        }
+      : undefined;
+    const webIdToSetPermsTo = `${permissionState.podUrlToSetPermissionsTo}profile/card#me`;
+
+    try {
+      await setDocContainerAclPermission(session, permissions, podUrl, webIdToSetPermsTo);
+
+      runNotification(
+        `${permissions.read ? 'Give' : 'Revoke'} permission to ${
+          permissionState.podUrlToSetPermissionsTo
+        } for Documents Container.`,
+        5,
+        state,
+        dispatch
+      );
+    } catch (error) {
+      runNotification('Failed to set permissions. Reason: File not found.', 5, state, dispatch);
+    } finally {
+      setTimeout(() => {
+        clearInputFields();
+      }, 3000);
+    }
+  };
+
+  return (
+    <Dialog open={showModal} onClose={handleCloseModal}>
+      <FormSection
+        title="Permission for Container"
```

Comment:
While it's correct that's it's permission for a container on Solid, think it'll be clearer if we specific what, like "Permissions to Documents"